### PR TITLE
BFD-2332: Update existing percentile CloudWatch SLO Alarms to not evaluate with a small sample size

### PR DESCRIPTION
**Pull request by malessi**
**JIRA Ticket:**
[BFD-2332](https://jira.cms.gov/browse/BFD-2332)

**User Story or Bug Summary:**

As a BFD/on-call Engineer, I would like the alerts received from percentile-based SLO alarms to be
accurate such that the alerts are actionable and indicate some sort of anomalous behavior.

---

### What Does This PR Do?

This PR updates all of the percentile-based CloudWatch SLO Alarms in the `bfd_server_slo_alarms`
Terraform module to specify that they should not evaluate if there is a low number of samples. This
should help to alleviate false positive alarming that we have seen from time-to-time with these
alarms as they would often alarm with too few samples.

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

- Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

- If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

  - Does this PR add any new software dependencies?
    - [ ] Yes
    - [x] No
  - Does this PR modify or invalidate any of our security controls?
    - [ ] Yes
    - [x] No
  - Does this PR store or transmit data that was not stored or transmitted before?
    - [ ] Yes
    - [x] No

- If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
  - Do you think this PR requires additional review of its security implications for other reasons?
    - [ ] Yes
    - [x] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

- N/A

### Submitter Checklist

I have gone through and verified that...:

- [x] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
- [x] This PR is reasonably limited in scope, to help ensure that:
  1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
  2. There isn't too much of a burden on reviewers.
  3. Any problems it causes have a small "blast radius".
  4. It'll be easier to rollback if that becomes necessary.
- [x] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
- [x] The data dictionary has been updated with any field mapping changes, if any were made.
- [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
- [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
- [x] Reviews are requested from both:
  - At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
  - Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
- [x] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
  - Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: ops/terraform/modules/resources/bfd_server_slo_alarms/main.tf [hunk 1]
```diff
@@ -139,12 +139,13 @@ resource "aws_cloudwatch_metric_alarm" "slo_coverage_bulk_latency_99p_15m_alert"
...
-  evaluation_periods  = "1"
-  period              = "900"
-  extended_statistic  = "p99"
-  threshold           = local.partners.bulk[each.key].timeout_ms
+  alarm_name                            = "${local.app}-${var.env}-slo-coverage-bulk-latency-99p-15m-alert-${each.key}"
+  comparison_operator                   = "GreaterThanThreshold"
+  evaluation_periods                    = "1"
+  period                                = "900"
+  extended_statistic                    = "p99"
+  evaluate_low_sample_count_percentiles = "ignore"
```
Comment (malessi) [inline ops/terraform/modules/resources/bfd_server_slo_alarms/main.tf:147]: FYI for reviewers: Git is showing a much larger diff due to running `terraform fmt` after making these changes. Of each diff, only a single line has been added:

```
evaluate_low_sample_count_percentiles = "ignore"
```

**Review by lsmitchell — APPROVED** [review]
Just curious what it considers a low number of samples? My concern is mostly that the frequency we report doesnt fall into the 'too-low' threshhold per timescale and thusly never reports anything.

**Comment by malessi** [comment]
> Just curious what it considers a low number of samples? My concern is mostly that the frequency we report doesnt fall into the 'too-low' threshhold per timescale and thusly never reports anything.

According to [the documentation](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/AlarmThatSendsEmail.html#percentiles-with-low-samples):

> For percentiles between 0.5 (inclusive) and 1.00 (exclusive), this setting is used when there are fewer than 10/(1-percentile) data points during the evaluation period. For example, this setting would be used if there were fewer than *1000 samples for an alarm on a p99 percentile*.

So, from this it's given that the percentile will not evaluate _until_ we receive 1000 samples. I say _until_ because of the following line in the documentation (emphasis mine):

> Or, you can have the alarm ignore the metric while the sample size is low, and *wait to evaluate it* until there is enough data to be statistically significant.

Given this I believe even if we do not receive 1000 datapoints (requests for a given endpoint) within a 15 minute period, _eventually_ this threshold will be crossed and the alarm *will* evaluate. So, assuming there is no surprising behavior here, the alarms should always be reporting something--it just might take longer than the typical 15 minute period for some less popular endpoints.

**Review by mjburling — APPROVED** [review]

**Review by mjburling — APPROVED** [review]

**Comment by mjburling** [comment]
Deployed to `test` without issue. This is ready to go. Thank, @malessi!

**Review by keithdadkins — APPROVED** [review]

**Pull request merged**